.circleci/config.yml

```diff
@@ -1,34 +1,59 @@
 version: 2
+
+build-template: &build-template
+  environment:
+    TEST_RESULTS: /tmp/test-results # path to where test results will be saved
+
+  steps:
+    - checkout
+    - run: mkdir -p $TEST_RESULTS # create the test results directory
+
+    - restore_cache: # restores saved cache if no changes are detected since last run
+        keys:
+          - go-pkg-mod-{{ checksum "go.sum" }}
+
+    - run:
+        name: "Precommit and Coverage Report"
+        command: |
+          make ci
+          mv coverage.html $TEST_RESULTS/
+
+    - save_cache:
+        key: go-pkg-mod-{{ checksum "go.sum" }}
+        paths:
+          - "/go/pkg/mod"
+
+    - store_artifacts:
+        path: /tmp/test-results
+        destination: opentelemetry-go-test-output
+
+    - store_test_results:
+        path: /tmp/test-results
+
 jobs:
-  build:
+  "go-1.14":
     docker:
       - image: circleci/golang:1.14
+    <<: *build-template
 
-    environment:
-      TEST_RESULTS: /tmp/test-results # path to where test results will be saved
+  "go-1.13":
+    docker:
+      - image: circleci/golang:1.13
+    <<: *build-template
 
+  build:
+    docker:
+      - image: circleci/golang:1.14
```

Comment:
It looks like the `build` stanza doesn't need the image here, is that right? (Not that it hurts.)

Reply:
I have been reading the CircleCI docs and every job needs to specify an executor (docker, machine, macos or windows). So even if we want an empty build, we need to have an executor. And docker executor requires specifying the image name.

So maybe instead of having three jobs (build, go-1.13 and go-1.14) how about we have two jobs - build and old-go-build? The build job could depend on the old-go-build job. That way, if the new version of go is released, all we need to update is docker image names. Job names will stay the same. Dependencies stay the same.